packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -190,8 +197,33 @@ export class ChainService implements ChainServiceInterface {
       to: nitroAdjudicatorAddress,
     };
 
-    const captureExpectedErrors = (reason: any) => {
-      if (!reason.error.message.includes('revert Channel finalized')) throw reason;
+    const captureExpectedErrors = async (reason: any) => {
+      if (reason.error?.message.includes('Channel finalized')) {
+        return;
+      }
+      const firstState = finalizationProof[0];
+      const [
+        _turnNumRecord,
+        finalizesAt,
+        _fingerprint,
+      ] = await this.nitroAdjudicator.getChannelStorage(
+        getChannelId({
+          ...firstState,
+          participants: firstState.participants.map(p => p.signingAddress),
+        })
+      );
+
+      // Check if the channel has been finalized in the past or will finalize in the next 10 minutes
+      const timeNowSeconds = Date.now() / 1_000;
```

Comment:
This should be `block.timestamp` shouldn't it? Who's to say whether `Date.now()` is aligned with the important timestamp?

Reply:
Yep, this is a great point 9d1de30